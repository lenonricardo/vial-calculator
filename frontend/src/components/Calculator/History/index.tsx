import {
  CustomDeleteForeverIcon,
  Expression,
  Total,
  Wrapper,
} from './styles'
import React, { useState, useEffect } from 'react'


interface History {
  expression: string,
  total: string,
}

export default function History() {

  const [history, setHistory] = useState<Array<History>>([])

  useEffect(() => {
    const operationHistory = getHistory()

    setHistory(operationHistory?.slice(0, 4) || [])
  }, [])

  const getHistory = () => {
    const storedHistory: string | null = localStorage.getItem('history')

    if (storedHistory) {
      return JSON.parse(storedHistory)
    }
  }

  const clearHistory = () => {
    localStorage.removeItem('history')
    setHistory([])
  }

  return (
    <Wrapper>
      <div>
        {history.map((result, i) => (
          <>
            <Expression key={i}>{result.expression} =</Expression>
            <Total key={i}>{result.total}</Total>
          </>
        ))}
      </div>

      {
        !!history.length &&
        <CustomDeleteForeverIcon onClick={clearHistory}/>
      }
    </Wrapper>
  )
}
