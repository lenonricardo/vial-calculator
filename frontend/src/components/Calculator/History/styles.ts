import styled from 'styled-components'
import DeleteForeverIcon from '@mui/icons-material/DeleteForever'

export const Wrapper = styled.div`
  height: var(--calculator-width);
  background: var(--secondary-operations);
  display: flex;
  flex-direction: column;
  border-radius: 0 0 1.5rem 1.5rem;
  padding-top: 1rem;
  align-items: flex-end;
  justify-content: space-between;
`

export const Expression = styled.span`
  width: 100%;
  display: flex;
  justify-content: flex-end;
  padding-right: 1.5rem;
  font-size: var(--memory-pad-font-size);
  color: var(--font-primary);
`
export const Total = styled.span`
  width: 100%;
  display: flex;
  justify-content: flex-end;
  padding: 0 1.5rem 2.5rem 0;
  font-size: var(--default-font-size);
  color: var(--font-primary);
`

export const CustomDeleteForeverIcon = styled(DeleteForeverIcon)`
  && {
    font-size: var(--default-font-size);
    color: var(--font-primary);
    cursor: pointer;
    margin-right: 1.5rem;
    margin-bottom: 1.5rem;
  }
`
