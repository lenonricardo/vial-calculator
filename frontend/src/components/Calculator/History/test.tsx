import { render } from '@testing-library/react'

import History from '.'

describe('<History />', () => {
  it('should render the component', () => {
    const { container } = render(<History />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
