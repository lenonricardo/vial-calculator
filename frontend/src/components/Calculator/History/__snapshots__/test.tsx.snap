// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<History /> should render the component 1`] = `
.c0 {
  height: var(--calculator-width);
  background: var(--secondary-operations);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border-radius: 0 0 1.5rem 1.5rem;
  padding-top: 1rem;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

<div
  class="c0"
>
  <div />
</div>
`;
